Source/WebCore/platform/graphics/ImageBufferContextSwitcher.cpp

```diff
@@ -82,9 +82,11 @@ void ImageBufferContextSwitcher::endClipAndDrawSourceImage(GraphicsContext& dest
 
 void ImageBufferContextSwitcher::endDrawSourceImage(GraphicsContext& destinationContext, const DestinationColorSpace& colorSpace)
 {
+    ASSERT_IMPLIES(m_filter, m_sourceImage);
```

Comment:
Why add this confusing conditional assertion? Why not just assert `m_sourceImage` after the `if` statement?

Also, the `USE(CAIRO)` case below does a null check of `m_sourceImage`. Why? That doesn’t seem right.

Reply:
A filter may not have a sourceImage. For example this filter does not require a sourceImage because it just fills a rectangle with color. 
```
<svg xmlns="http://www.w3.org/2000/svg" width="200" height="200">
    <filter id="floodFilter" filterUnits="userSpaceOnUse">
      <feFlood x="50" y="50" width="100" height="100" flood-color="green" />
    </filter>
  <use style="filter: url(#floodFilter);" />
</svg>
```

These are the four cases of `m_filter` and `m_sourceImage` in this cases and the meanings of these cases:
```
m_filter    m_sourceImage   Meaning
---------------------------------------------------
nullptr     nullptr         This is an error like the sourceImageRect is too big so the m_sourceImage can't be allocated. We should handle it safely or RELEASE_ASSERT.
nullptr     VALID           This is correct. It happens with canvas layers. This class is used to switch the rendering to anther temporary bitmap with and without a filer.
VALID       nullptr         This the case of a filter without a sourceImage <feFlood> for example
VALID       VALID           This the case of a filter with a sourceImage <feDropShadow> for example
```